osquery/tables/system/darwin/sysctl_utils.cpp

```diff
@@ -41,6 +46,58 @@ const std::vector<std::string> kControlNames{
 const std::vector<std::string> kControlTypes{
     "", "node", "int", "string", "quad", "opaque", "struct"};
 
+void opaquePushback(QueryData& results,
+                      Row& r,
+                      std::string opaque_cv,
+                      const std::string& var) {
+      r["field_name"] = var;
+      r["current_value"] = opaque_cv;
+      results.push_back(r);
+  };
+
+void opaqueControlInfo(QueryData& results,
+                       Row& r,
+                       char* response,
+                       std::string& value) {
+  if (value.compare("S,clockinfo") == 0) {
+      struct clockinfo *ci = (struct clockinfo*) response;
+      opaquePushback(results, r, INTEGER(ci->hz), "hz");
+      opaquePushback(results, r, INTEGER(ci->tick), "tick");
+      opaquePushback(results, r, INTEGER(ci->tickadj), "tickadj");
+      opaquePushback(results, r, INTEGER(ci->profhz), "profhz");
+      opaquePushback(results, r, INTEGER(ci->stathz), "stathz");
+    }
+  else if (value.compare("S,timeval") == 0) {
+      struct timeval *tv = (struct timeval*) response;
+      opaquePushback(results, r, INTEGER((long)tv->tv_sec), "sec");
+      opaquePushback(results, r, INTEGER((long)tv->tv_usec), "usec");
+    }
+  else if (value.compare("S,loadavg") == 0) {
+      struct loadavg *tv = (struct loadavg*) response;
+      opaquePushback(results, r,
+        DOUBLE((double)tv->ldavg[0]/(double)tv->fscale), "ldavg0");
+      opaquePushback(results, r,
+        DOUBLE((double)tv->ldavg[1]/(double)tv->fscale), "ldavg1");
+      opaquePushback(results, r,
+        DOUBLE((double)tv->ldavg[2]/(double)tv->fscale), "ldavg2");
+    }
+  else if (value.compare("S,xsw_usage") == 0) {
+      struct xsw_usage *xsu = (struct xsw_usage*) response;
+      opaquePushback(results, r,
+        DOUBLE((double)xsu->xsu_total/(1024.0 * 1024.0)), "xsu_total");
```

Comment:
Use a names casts instead of C-style cast, please : [ref to CppCoreGuidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es49-if-you-must-use-a-cast-use-a-named-cast)